assets/js/react/GameStateEditor/components/EntityForm/EntityForm.jsx

```diff
@@ -184,86 +306,68 @@ class EntityForm extends Component {
           <input
             required={true}
             className="form-input"
-            value={entity.id}
+            value={id}
             onChange={this.handleChange}
             type="text"
             name="id"
           />
         </div>
-        <div className="form-row">
-          <label required={true} className="form-label">
-            Description
-          </label>
-          <textarea
-            required={true}
-            className="form-input"
-            onChange={this.handleChangeProps}
-            value={entity.props.description}
-            name="description"
-          />
-        </div>
-        <div className="form-row">
-          <label className="form-label">Behaviors</label>
-          <Select
-            name="behaviors"
-            className="form-input"
-            multi={true}
-            removeSelected={true}
-            value={entity.behaviors}
-            options={behaviorsForSelect}
-            onChange={this.handleChangeBehaviors}
-          />
-        </div>
         <div className="form-row">
           <label className="form-label">Location</label>
           <Select
-            name="location"
-            className="form-input"
-            value={entity.props.location}
-            options={locationsForSelect}
-            onChange={this.handleChangePropsDropdown("location")}
-          />
-        </div>
-        <div className="form-row">
-          <label className="form-label">Destination</label>
-          <Select
-            name="destination_id"
+            name="props.location"
             className="form-input"
-            value={entity.props.destination_id}
+            value={props.location}
             options={locationsForSelect}
-            onChange={this.handleChangePropsDropdown("destination_id")}
+            onChange={this.handleChangeLocation}
           />
         </div>
+        {this.renderBehaviors()}
+        {this.renderAddNewBehavior()}
         <div className="form-actions">
-          <button disabled={!this.valid()} onClick={this.handleAdd}>
-            Add
+          <button
+            disabled={!this.valid()}
+            onClick={this.handleAdd}
+            className="button"
+          >
+            Add/Update
+          </button>
+          <button onClick={this.handleClear} className="button">
+            Clear
           </button>
-          <button onClick={this.handleClear}>Clear</button>
+        </div>
+        <div className="EntityForm__state">
```

Comment:
This can be removed at some point.  Until we know this is really working, it is nice to see the react state as we buildl it up with the form.